### ISSUE TITLE
Form field `defaultValue` is not displaying as value or placeholder

### ISSUE THREAD

**Issue by seanconklin** [Automattic/jetpack#14305]
#### Steps to reproduce the issue

Prerequisite: Have Jetpack enabled.

1. Go to WP Admin > Pages > Add New (with Block Editor mode)
2. Click on (+) to add a new block
3. Type "Form" and select the Form block.
4. Fill in email address and subject line and click Add Form button.
5. In each of the default fields that appear (Name, Email, URL, Message) click them and provide a fitting value. To do this, use the right sidebar (or click the gear icon to toggle it on) > Block > Field Settings > Default Value field.
6. Click Preview button.
7. Observe the form has no values in any of the fields.

#### What I expected
The form fields to have default values displayed.

#### What happened instead
Form fields are empty, without a value or a placeholder.

I clicked into the Code Editor to confirm my entries were saved, and I observed they are. Here's my form code:

```
<!-- wp:jetpack/contact-form {"subject":"TEST","to":"tester@test.com","hasFormSettingsSet":"yes"} -->
<!-- wp:jetpack/field-name {"required":true,"defaultValue":"Tester"} /-->

<!-- wp:jetpack/field-email {"required":true,"defaultValue":"test@test.com"} /-->

<!-- wp:jetpack/field-url {"defaultValue":"https://test.com"} /-->

<!-- wp:jetpack/field-textarea {"defaultValue":"This is a test"} /-->
<!-- /wp:jetpack/contact-form -->
```

**Comment by seanconklin** [comment]
I reproduced this issue with both the native Gutenberg and with the Gutenberg plugin enabled. It seems to be the block within JetPack itself.

**Comment by jeherve** [comment]
Thanks for the report. I believe this is a duplicate of this issue: #12298 

I'll consequently close this one as a duplicate.
